lms/models/course.py

```diff
@@ -0,0 +1,77 @@
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.mutable import MutableDict
+from sqlalchemy.orm.exc import NoResultFound
+
+from lms.db import BASE
+from lms.models.application_settings import ApplicationSettings
+
+
+class Course(BASE):
+    """An LTI course."""
+
+    __tablename__ = "course"
+
+    #: The authority_provided_id that uniquely identifies the course that these
+    #: settings belong to.
+    authority_provided_id = sa.Column(sa.UnicodeText(), primary_key=True)
+
+    #: The LTI consumer_key (oauth_consumer_key) of the application instance
+    #: that these course settings belong to.
+    consumer_key = sa.Column(
+        sa.String(),
+        sa.ForeignKey("application_instances.consumer_key", ondelete="cascade"),
+        primary_key=True,
+    )
```

Comment:
Why don't we use the primary key ("id") of the application instance here? / Why does an application instance have one if the consumer key is the de-facto primary key?

Reply:
I copied what we'd done elsewhere (in `models.oauth2_token`), but I think we want to be able to easily look up a `models.Course` by `request.lti_user.consumer_key` and `context.h_group.authority_provided_it`, and that's more convenient if `consumer_key` is directly on the `course` table rather than in a related table.

I dunno why `application_instance` has an `id` and doesn't just use the `consumer_key` as its primary key

Reply:
Ah well. I wonder if we could just drop the column if it isn't actually used? (Not for now obvs)